src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+        // Outputs
+
+        // Transition
+        nextState = SuperstructureStates.IDLE
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedUpAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.groundIntakeConeExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE -> {
+        // Outputs
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.groundIntakeConeExtension.get(),
+            Manipulator.TunableManipulatorStates.coneInVoltage.get()
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP -> {
+        // Outputs
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.minExtension.get(),
+            Manipulator.TunableManipulatorStates.coneIdleVoltage.get()
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.DOUBLE_SUBSTATION_INTAKE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedDownAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.doubleSubstationIntakeShelfExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+          val offset =
+            when (usingGamePiece) {
+              GamePiece.CUBE -> Elevator.TunableElevatorHeights.shelfIntakeCubeOffset.get()
```

Comment:
I'm wondering if we actually get anything out of this cube and cone offset thing other than an extra `TunableNumber`. In terms of tuning quickly, it seems that we could just change this to 2 tunable numbers: `doubleSubstationCubeHeight` and `doubleSubstationConeHeight` and that would be easier to tune tbh -- because then you just get the elevator to the height you want (while disabled, lift the carriage) and write down one number, as opposed to a number and an offset or whatever